packages/waku/src/lib/renderers/rsc-renderer.ts

```diff
@@ -188,7 +198,46 @@ export async function renderRsc(
     ) {
       // XXX This doesn't support streaming unlike busboy
       const formData = parseFormData(bodyStr, contentType);
-      args = await decodeReply(formData);
+      const moduleMap = new Proxy({} as Record<string, ImportManifestEntry>, {
+        get(_target, rsfId: string): ImportManifestEntry {
+          const [fileId, name] = rsfId.split('#') as [string, string];
+          // fixme: race condition, server actions are not initialized in the first time
+          if (!moduleLoading.has(fileId)) {
+            if (!opts.isDev) {
+              moduleLoading.set(
+                fileId,
+                import(
+                  /* @vite-ignore */
+                  fileId
+                ).then((m: any) => {
+                  moduleCache.set(
+                    fileId,
+                    Object.fromEntries(m['__waku_serverActions']),
+                  );
+                }),
+              );
+            } else {
+              moduleLoading.set(
+                fileId,
+                opts
+                  .loadServerFile(filePathToFileURL(fileId))
+                  .then((m: any) => {
+                    moduleCache.set(
+                      fileId,
+                      Object.fromEntries(m['__waku_serverActions']),
+                    );
+                  }),
+              );
+            }
+          }
+          return {
+            id: fileId,
+            chunks: [],
```

Comment:
`chunks: [fileId]` is not necessary after reading the source code fo next.js

Reply:
Are you sure? I mean waku's approach might be different from nextjs's one.